File: app/src/main/java/catchla/yep/graphic/ActionBarColorDrawable.java
/*
 * Copyright (c) 2015. Catch Inc,
 */

package catchla.yep.graphic;

import android.annotation.TargetApi;
import android.graphics.Outline;
import android.graphics.Rect;
import android.os.Build;

/**
 * Created by mariotaku on 14/12/8.
 */
public class ActionBarColorDrawable extends ActionBarColorDrawableBase {
    public ActionBarColorDrawable(boolean outlineEnabled) {
        super(outlineEnabled);
    }

    public ActionBarColorDrawable(int color, boolean outlineEnabled) {
        super(color, outlineEnabled);
    }

    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    @Override
    public void getOutline(Outline outline) {
        if (!isOutlineEnabled()) return;
        final Rect bounds = getBounds();
        // Very very dirty hack to make outline shadow in action bar not visible beneath status bar
        outline.setRect(bounds.left - bounds.width() / 2, -bounds.height(),
                bounds.right + bounds.width() / 2, bounds.bottom);
        outline.setAlpha(getAlpha() / 255f);
    }
}

File: src/main/java/catchla/yep/graphic/ActionBarColorDrawable.kt
/*
 * Copyright (c) 2015. Catch Inc,
 */

package catchla.yep.graphic

import android.annotation.TargetApi
import android.graphics.Outline
import android.os.Build

/**
 * Created by mariotaku on 14/12/8.
 */
class ActionBarColorDrawable : ActionBarColorDrawableBase {
    constructor(outlineEnabled: Boolean) : super(outlineEnabled) {
    }

    constructor(color: Int, outlineEnabled: Boolean) : super(color, outlineEnabled) {
    }

    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    override fun getOutline(outline: Outline) {
        if (!isOutlineEnabled) return
        val bounds = bounds
        // Very very dirty hack to make outline shadow in action bar not visible beneath status bar
        outline.setRect(bounds.left - bounds.width() / 2, -bounds.height(),
                bounds.right + bounds.width() / 2, bounds.bottom)
        outline.alpha = alpha / 255f
    }
}
